### DISCUSSION TITLE
z-index on Spotfire mods

### DISCUSSION THREAD

**Discussion by CommonCrisis** [TIBCOSoftware/spotfire-mods#108] [Q&A]
Hey there,

is it possible to let spotfire mods pop on top of the whole page?
Using z-index doesn't seem to work and the popup API is really limited to what I can display...

**Comment by objerke** [comment]
Hi @CommonCrisis,

Visualization Mods are currently confined within its visual boundaries on the canvas. Could you elaborate on your use case?

**Reply by CommonCrisis** [reply]
I'd really like to use different react MUI components as mods in the Spotfire reports - that way I can build customized components with version control etc. accessing the data loaded into my report.

E.g. a selection data load stepper or a simple config button that pops up and let's the user config the visualization, giving the non technical user the experience he or she is used to in the web.

Then I could use the visualizations and backend by SPotfire but I can replace the sometimes really "ugly" and clunky UI elements Spotfire provides with good looing ones.

Hope my use case is clear...

If would have the freedom to use the Popup API with what ever I want to render into it, that would be fine for me.

**Reply by ayh20** [reply]
There are a number of example mods that have pop up config panels. 

It's done as part of the mod within the same container. This is an example of one i did .... gear icon ... click to get popup menu ...
![clockmenu](https://user-images.githubusercontent.com/17643158/233355334-78d316a8-3db0-4381-8200-7508b38e3365.png)

**Reply by CommonCrisis** [reply]
> There are a number of example mods that have pop up config panels.
> 
> It's done as part of the mod within the same container. This is an example of one i did .... gear icon ... click to get popup menu ... ![clockmenu](https://user-images.githubusercontent.com/17643158/233355334-78d316a8-3db0-4381-8200-7508b38e3365.png)

Yes but if your visualization would be way smaller your popup could not break out of the visualization boundaries. So you would not see the whole thing.

**Comment by mrylande** [comment]
Hi @CommonCrisis,

What you are asking for is currently not supported. Please add your improvement suggestions to the [TIBCO Ideas Portal](https://ideas.tibco.com). More info can be found [here](https://community.tibco.com/s/article/Introduction-to-the-TIBCO-Ideas-Portal).
